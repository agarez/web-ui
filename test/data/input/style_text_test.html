<!doctype html>
<!--
Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="packages/web_ui/testing/testing.js"></script>
</head>
<body>
  Verify that style works with text bindings:
  <div id="test1" style="color: {{color}}; height: {{height}}px;">
  this div will have blue (0000ff) font color and 50px height.
  </div>
  <script type="application/dart">
    @observable
    library main;
    import 'dart:async';
    import 'dart:html';
    import 'package:web_ui/web_ui.dart';
    import 'package:unittest/unittest.dart';

    var color = 'blue';
    var height = 50;
    main() {
      Timer.run(() {
        expect(query('#test1').style.color, color);
        expect(query('#test1').style.height, '${height}px');
        window.postMessage('done', '*');
      });
    }
  </script>
</body>
</html>
