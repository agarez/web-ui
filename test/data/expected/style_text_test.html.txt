Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x84
  RenderBlock {HTML} at (0,0) size 800x84
    RenderBody {BODY} at (8,8) size 784x68
      RenderBlock (anonymous) at (0,0) size 784x18
        RenderText {#text} at (0,0) size 430x18
          text run at (0,0) width 430: "Verify that style works with text bindings:"
      RenderBlock {DIV} at (0,18) size 784x50 [color=#0000FF]
        RenderText {#text} at (0,0) size 600x18
          text run at (0,0) width 600: "this div will have blue (0000ff) font color and 50px height."
